core/src/util/fastmap.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
+#include <vector>
+#include <string>
+#include <algorithm>
+
+namespace Tangram {
+
+// see also:
+// http://www.boost.org/doc/libs/1_59_0/doc/html/boost/container/flat_map.html
+// https://realm.io/assets/news/binary-search/blog.cpp
+
+template<typename K, typename T>
+struct fastmap {
+    std::vector<std::pair<K, T>> map;
```

Comment:
I would stick with fastmap for now as it showed good improvements in perf.
Yes benchmarks are needed, but I don't see any point in postponing these
changes until benchmarks have been written.

Matt's suggestion could definitely be further improvement on this.
On Oct 21, 2015 2:38 AM, "Hannes Janetzek" <notifications@github.com> wrote:

> In core/src/util/fastmap.h
> <https://github.com/tangrams/tangram-es/pull/357#discussion_r42588127>:
>
> > @@ -0,0 +1,115 @@
> > +#pragma once
> > +#include <vector>
> > +#include <string>
> > +#include <algorithm>
> > +
> > +namespace Tangram {
> > +
> > +// see also:
> > +// http://www.boost.org/doc/libs/1_59_0/doc/html/boost/container/flat_map.html
> > +// https://realm.io/assets/news/binary-search/blog.cpp
> > +
> > +template<typename K, typename T>
> > +struct fastmap {
> > +    std::vector<std::pair<K, T>> map;
>
> Yes, we should try that. It probably depends on the size of the map. This
> is is a first start which showed clear improvement in manual perf testing.
> But we could also postpone fastmap until we have a proper benchmarks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tangrams/tangram-es/pull/357/files#r42588127>.
>